src/mds/Server.cc

```diff
@@ -3839,7 +3839,8 @@ void Server::handle_client_setattr(MDRequestRef& mdr)
 								   truncating_smaller, changed_ranges));
 
   // flush immediately if there are readers/writers waiting
-  if (cur->get_caps_wanted() & (CEPH_CAP_FILE_RD|CEPH_CAP_FILE_WR))
+  if (xlocks.count(&cur->filelock) &&
+      (cur->get_caps_wanted() & (CEPH_CAP_FILE_RD|CEPH_CAP_FILE_WR)))
```

Comment:
True. But we still want to flush if any other clients are waiting, not just the requesting client (who, after all, can make progress without the change being durably recorded). So we can be more precise than just looking at get_caps_wanted(), but I think checking if we xlocked is too specific?